test/e2e/util/utils.go

```diff
@@ -391,6 +416,169 @@ func DeployEraserHelm(namespace string, args ...string) env.Func {
 	}
 }
 
+func GetManagerLogs(ctx context.Context, cfg *envconf.Config, t *testing.T) error {
+	c, err := cfg.NewClient()
+	if err != nil {
+		return err
+	}
+
+	var pods corev1.PodList
+	err = c.Resources().List(ctx, &pods, func(o *metav1.ListOptions) {
+		o.LabelSelector = labels.SelectorFromSet(map[string]string{"control-plane": "controller-manager"}).String()
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(pods.Items) > 1 {
+		return errors.New("only one manager pod should be present")
+	} else if len(pods.Items) == 0 {
+		return errors.New("no manager pod present")
+	}
+
+	manager := pods.Items[0]
+
+	output, err := KubectlLogs(cfg.KubeconfigFile(), manager.Name, "", EraserNamespace)
+	if err != nil {
+		return err
+	}
+
+	testName := strings.Split(t.Name(), "/")[0]
+
+	// get log output file path
+	path := filepath.Join(TestLogDir, testName)
+
+	var file *os.File
+	if err := os.MkdirAll(path, filemode); err != nil {
+		return err
+	}
+	_, err = os.Create(filepath.Join(path, manager.Name))
+	if err != nil {
+		return err
+	}
+
+	file, err = os.OpenFile(filepath.Join(path, manager.Name), os.O_APPEND|os.O_WRONLY, filemode)
```

Comment:
So before `filemode` was set to `0644` which is good for `os.OpenFile`.
It allows read/write for the user that created it, and read-only for users in the same group and everyone else.

Now `filemode is `0755` which gives execute permissions and other things that directories need.

I think I would just explicitly set `0644` in `OpenFile` and `0755` for `Mkdir` rather than using a constant.